sympy/functions/combinatorial/numbers.py

```diff
@@ -988,7 +988,9 @@ def _eval_rewrite_as_tractable(self, n, m=1, limitvar=None, **kwargs):
         pg = self.rewrite(polygamma)
         if not isinstance(pg, harmonic):
             return pg.rewrite("tractable", deep=True)
-        return (zeta(m) - zeta(m, n+1)).rewrite("tractable", deep=True)
+        arg = m - S.One
+        if arg.is_positive or arg.is_negative:
```

Comment:
Could this be `(m - 1).is_nonzero`?

Reply:
Yeah , might have missed this in a hurry !

Reply:
I have a little diff that will implement the asymptotic series expansion of `zeta(s,a)` with respect to `a` when the latter is a positive integer greater than 1. However it has to be done _after_ #24012 is merged – that is a big change and its new stuff will likely offer additional opportunities for the `zeta.aseries()` PR.

Reply:
> However it has to be done after https://github.com/sympy/sympy/pull/24012 is merged

Sure, but as you've mentioned , even I feel that we might be able to address a better range of cases once that is implemented fully !